aspnetcore/native-aot/index.md

```diff
@@ -0,0 +1,179 @@
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <!-- Other properties ommitted for brevity -->
+    <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
+  </PropertyGroup>
+
+</Project>
+```
+
+With the project file updated, run the `dotnet build` command (`publish` isn't necessary to view generated code). In the built output file there will be a `obj/Debug/net8.0/generated/` directory which contains all the generated files for the project.
+
+When `dotnet publish` is run, the project source files and generated source files are compiled as normal and then outputted assemblies are passed into an native IL compiler which produces the native executable which contains the native machine code to run the application.
+
+
+## Benefits of using .NET native AOT with ASP.NET Core
+
+Using the .NET native AOT deployment model provides the following benefits:
+
+** **Minimize disk footprint**: When publishing using native AOT a single executable is produced containing just the code from external dependencies that is used to support the program. Reduced executable size can lead to:
+  * Smaller container images, for example in containerized deployment scenarios.
+  * Reduce deployment time from smaller images.
+* **Reduced startup time**: Native AOT applications can show reduced start-up times. Reduced start-up means:
+  * The app is ready to service requests quicker.
+  * Improved deployment where container orchestrators need manage transition from one version of the app to another.
+* **Reduce memory demand**: Native AOT apps can have reduced memory demands depending on the work being performed by the app. Reduced memory consumption can lead to greater deployment density and improved scalability.
+
+
+## ASP.NET Core and AoT Supportability
+
+Not all features in ASP.NET Core are currently compatible with .NET native AOT. The following table summarizes ASP.NET Core feature compatibility with .NET native AOT:
+
+| Feature | Fully Supported | Partially Supported | Not Supported |
+| - | - | - | - |
+| gRPC | <span aria-hidden="true">✔️</span><span class="visually-hidden">Fully supported</span> | | |
+| Minimal APIs | | <span aria-hidden="true">✔️</span><span class="visually-hidden">Partially supported</span> | |
+| MVC | | | <span aria-hidden="true">❌</span><span class="visually-hidden">Not supported</span> |
+| Blazor Server | | |<span aria-hidden="true">❌</span><span class="visually-hidden">Not supported</span> |
+| SignalR | | | <span aria-hidden="true">❌</span><span class="visually-hidden">Not supported</span> |
+| Authentication | | | <span aria-hidden="true">❌</span><span class="visually-hidden">Not supported</span> (JWT soon) |
```

Comment:
Authentication is one middleware of many. The vast majority of other middleware should work.

I think we should go ahead and include them:

* CORS
* HealthChecks
* HttpLogging
* Localization (test this one first)
* OutputCaching
* RateLimiting
* RequestDecompression
* ResponseCaching
* ResponseCompression
* Rewrite
* Session (put this one under not supported, uses Cryptograph.Xml)
* Spa (put this one under not supported, it's not annotated)
* StaticFiles
* WebSockets

Reply:
I don't think localization will work because we actually disable it in the template properties (we just use invariant culture).